Index: setup.py
===================================================================
--- setup.py	(revision 136)
+++ setup.py	(working copy)
@@ -167,6 +167,8 @@
         multi_perl = os.path.join(cur_dir, 'MULTI_PERL')
         if os.access(multi_perl, os.F_OK):
             os.unlink(multi_perl)
+        if self.root is None:
+            self.root = ''
 
         os.chdir('Python-Object')
         install.spawn(self, ["make", "DESTDIR=%s" % self.root, "install"])
